apiserver/facades/client/controller/controller.go

```diff
@@ -196,65 +198,116 @@ func (c *ControllerAPI) MongoVersion() (params.StringResult, error) {
 	return result, nil
 }
 
-// DashboardAddressInfo returns the address info needed to connect to the juju dashboard.
-func (c *ControllerAPI) DashboardAddressInfo() (params.DashboardInfo, error) {
-	getDashboardInfo := func() ([]string, bool, error) {
+// dashboardConnectionInforForCAAS returns a dashboard connection for a Juju
+// dashboard deployed on CAAS.
+func (c *ControllerAPI) dashboardConnectionInfoForCAAS(
+	model *state.Model,
+	applicationName string,
+) (*params.DashboardConnectionProxy, error) {
+	configGetter := stateenvirons.EnvironConfigGetter{Model: model}
+	environ, err := common.EnvironFuncForModel(model, configGetter)()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	caasBroker, ok := environ.(caas.Broker)
+	if !ok {
+		return nil, errors.New("unable to get cass environ for model")
```

Comment:
Generally errors should be phrased like:
`cannot get cars environ for model`
`cannot blah blah`